File: leetcode-submissions/1829_Maximum_XOR_for_Each_Query.cpp
class Solution {
public:
    vector<int> getMaximumXor(vector<int>& nums, int maximumBit) {
        // we need to find out the xor of all the elements
        int XOR=0,n=nums.size();
        vector<int>result(n);
        for(int ele:nums){
            XOR^=ele;
        }
        //after this we need to create a mask 
        int mask=((1<<maximumBit)-1);
        for(int i=0;i<n;i++){
            int k=XOR^mask;
            result[i]=k;
            XOR=(XOR^nums[n-i-1]);
        }
        return result; 
    }
};

File: leetcode-submissions/1829_Maximum_XOR_for_Each_Query.ts
function getMaximumXor(nums: number[], maximumBit: number): number[] {
    const result:number[] =[];
    let XOR=0;

    for(const ele of nums){
XOR^=ele;
    }
    let mask=((1<<maximumBit)-1);
    for(let i=0;i<nums.length;i++){
        let k=XOR^mask;
        result.push(k);
        XOR=(XOR^nums[nums.length-1-i]);
    }
    return result;
};